### ISSUE TITLE
Compiler error in Signature.c when building using gcc 10  (stringop-truncation)

### ISSUE THREAD

**Issue by nickbroon** [tumi8/vermont#146]
Building on Debian Bullseye, which uses GCC 10 by default.

```
[  132s] cd /usr/src/packages/BUILD/obj-x86_64-linux-gnu/src/modules && /usr/bin/cc -DDISABLE_UDP_CONNECT -DENABLE_VRF -DHAVE_BOOST_FILESYSTEM -DJOURNALD_SUPPORT_ENABLED -DPCAP_DEFAULT_CAPTURE_LENGTH=128 -DPCAP_MAX_CAPTURE_LENGTH=128 -DSUPPORT_NETFLOWV9 -DSUPPORT_SCTP -DZMQ_SUPPORT_ENABLED -I/usr/src/packages/BUILD/src -I/usr/include/mit-krb5 -I/usr/include/pgm-5.3 -I/usr/include/libxml2 -I/usr/include/uuid -I/usr/include/nss -I/usr/include/nspr -g -O2 -ffile-prefix-map=/usr/src/packages/BUILD=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -std=gnu11 -Werror -Wall -pedantic -Wextra -Wno-unused-parameter -o CMakeFiles/modules.dir/analysis/fpsigmatcher/Signature.c.o -c /usr/src/packages/BUILD/src/modules/analysis/fpsigmatcher/Signature.c
[  132s] In file included from /usr/include/string.h:495,
[  132s]                  from /usr/src/packages/BUILD/src/modules/analysis/fpsigmatcher/Signature.h:29,
[  132s]                  from /usr/src/packages/BUILD/src/modules/analysis/fpsigmatcher/Signature.c:21:
[  132s] In function 'strncpy',
[  132s]     inlined from 'add_classid_to_signatureclass' at /usr/src/packages/BUILD/src/modules/analysis/fpsigmatcher/Signature.c:204:2:
[  132s] /usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:10: error: '__builtin_strncpy' specified bound 50 equals destination size [-Werror=stringop-truncation]
[  132s]   106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
[  132s]       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[  132s] cc1: all warnings being treated as errors
[  132s] make[3]: *** [src/modules/CMakeFiles/modules.dir/build.make:436: src/modules/CMakeFiles/modules.dir/analysis/fpsigmatcher/Signature.c.o] Error 1
[  132s] make[3]: *** Waiting for unfinished jobs....

```

**Comment by nickbroon** [comment]
Doh, I was looking at the wrong branch. Should be fixed by 4c1b3f365f9da26993157a9e0f4f49bf4d53da05

**Comment by nickbroon** [comment]
Doh, I was looking at the wrong branch. Should be fixed by 4c1b3f365f9da26993157a9e0f4f49bf4d53da05
